misk-redis/src/main/kotlin/misk/redis/RedisClusterModule.kt

```diff
@@ -90,10 +90,15 @@ class RedisClusterModule @JvmOverloads constructor(
       //CLIENT SETINFO is only supported in Redis v7.2+
       .clientSetInfoConfig(ClientSetInfoConfig.DISABLED)
       .build()
+
+    // We want to support services running both under docker and localhost when running locally and this is a way to support that.
+    // If a hostname is provided, it will always take precedence over the environment variable.
+    val redisHost = replicationGroup.configuration_endpoint.hostname?.takeUnless { it.isNullOrBlank() } ?: System.getenv("REDIS_HOST") ?: "127.0.0.1"
```

Comment:
No using orc we can't just set the set the hostname to localhost for e.g. [here](https://github.com/squareup/cash-misk-exemplar-service/blob/main/service/src/main/resources/cash-misk-exemplar-service-orc.yaml#L12) given using orc it can either be `localhost` or `docker.host.internal` depending on how the application is started. Given misk config loader doesn't have a way unfurl the yaml file and look for envar as armeria config files [as such](https://github.com/squareup/cash-p2p-index/blob/c3981f6ebe169800d94bfd1084d73b23312038c5/service/src/main/resources/data-source-clusters-development.yaml#L6), we need to explicitly check for the envar this way in the framework code.

Reply:
We've followed a similar approach for sql as well for misk, and I encountered some of this when I tested this extensively at that time https://github.com/cashapp/misk/pull/3004. There is some more context in the comment thread and the linked slack threads.


Longer term I would like we adopt unfurling the configs like we do in armeria for misk configs as well.